openmdao/core/group.py

```diff
@@ -4813,6 +4813,23 @@ def _sorted_sys_iter(self):
         else:
             yield from self._subsystems_myproc
 
+    def _sorted_sys_iter_all_procs(self):
+        """
+        Yield subsystem names in sorted order if Problem option allow_post_setup_reorder is True.
+
+        Otherwise, yield subsystem names in the order they were added to their parent group.
+
+        Yields
+        ------
+        System
+            A subsystem.
+        """
+        if self._problem_meta['allow_post_setup_reorder']:
+            for s in sorted(self._subsystems_allprocs):
```

Comment:
Did it break something if you didn't sort these?  It doesn't seem like it should matter.

Reply:
Yeah, they weren't in the right order if I didn't sort them. The serial order also sorts them like this.